xbmc/guilib/GUIControl.h

```diff
@@ -370,6 +371,7 @@ class CGUIControl
   float m_stereo;
   TransformMatrix m_transform;
   TransformMatrix m_cachedTransform; // Contains the absolute transform the control
+  bool m_isTransparent;
```

Comment:
```c++
bool m_isTransparent{true};
```

and the constructors could be unmodified.

Reply:
I could compare the alpha of the cached transform each time to eliminate the bool. But thinking a bit more about the matter, I should rename it to m_isCulled. Then can also be used to signal if the element has been translated off screen by the animation.

From my understanding, we have three phases for controls.
1) Setup phase. Controls can be manipulated (by SetPosition() or similar) . That's done once on window load but might vary by the use of scripts.
2) Processing phase. Animations are calculated here. Dirty regions are also set here.
3) Rendering phase.

In phase 1, we don't know if the element is being rejected after the animation. So we have two options:
A) Don't mark it as dirty in phase 1 if it was not drawn in the last frame. If the state changes to true after the animation, it has to be marked as dirty in any case.
B) Mark it as being potentially dirty in phase 1 and do the actual marking via MarkDirtyRegion() in phase 2 after the animation.

I've gone with A, as it is the least invasive option. B would be technically correct, but would require more changes. Testing this would be a nightmare, as the whole dirty region rendering is a mess. In the end, B would only save a dirty region for one frame in case the control gets marked dirty in phase 1 but the animation rejects it at phase 2.

Reply:
Having an invalid initial state would result in the element being marked as dirty the first time after its creation, even if the animation might cull it.

Reply:
It seems like both A and B have downsides. Feel free to merge solution A with solution B as a "follow-up" (which may never be completed but will at least document how things should be)